roles/openshift_efk/tasks/main.yaml

```diff
@@ -0,0 +1,179 @@
+    register: fluentd_output
+    failed_when: "fluentd_output.rc == 1 and 'exists' not in fluentd_output.stderr"
+    tags:
+      - build
+
+  - name: "Set additional permissions for fluentd"
+    command: oadm policy add-cluster-role-to-user cluster-reader system:serviceaccount:logging:aggregated-logging-fluentd
+    register: fluentd2_output
+    failed_when: "fluentd2_output.rc == 1 and 'exists' not in fluentd2_output.stderr"
+    tags:
+      - build
+
+  - name: "Make sure to remove stale deployer template"
+    command: oc delete template logging-deployer-template -n openshift
+    register: delete_ouput
+    failed_when: delete_ouput.rc == 1 and 'exists' not in delete_ouput.stderr
+    tags:
+      - build
+
+  - name: "Create deployer template"
+    command: oc create -f /usr/share/openshift/examples/infrastructure-templates/enterprise/logging-deployer.yaml -n openshift
+    register: template_output
+    failed_when: "template_output.rc == 1 and 'exists' not in template_output.stderr"
+    tags:
+      - build
+
+  - name: "Clear out any previous pods"
+    command: oc delete pods --all
+    tags:
+      - build
+
+  - name: "Process the deployer template with an registry other than registry.access.redhat.com"
+    shell: oc process logging-deployer-template -n openshift -v KIBANA_HOSTNAME={{ kibana_hostname | quote }},ES_CLUSTER_SIZE={{ es_cluster_size | quote }},PUBLIC_MASTER_URL={{ master_url | quote }},IMAGE_PREFIX={{ target_registry | quote }}/  | oc create -f -
+    when: target_registry is defined
+    tags:
+      - build
+
+  - name: "Process the default deployer template"
+    shell: oc process logging-deployer-template -n openshift -v KIBANA_HOSTNAME={{ kibana_hostname | quote }},ES_CLUSTER_SIZE={{ es_cluster_size | quote }},PUBLIC_MASTER_URL={{ master_url | quote }}  | oc create -f -
+    when: target_registry is not defined
+    tags:
+      - build
+
+  - name: "Wait for image pull and deployer pod"
+    action: shell oc get pods | grep logging-deployer.*Completed
+    register: result
+    until: result.rc == 0
+    retries: 15
+    delay: 10
+    tags:
+      - build
+
+  - name: "Process support template"
+    shell: oc process logging-support-template | oc create -f -
+    tags:
+      - build
+
+  - name: "Set insecured registry"
+    command: oc annotate is --all  openshift.io/image.insecureRepository=true --overwrite
+    when: "target_registry is defined and insecure_registry == 'true'"
+    tags:
+      - build
+
+  - name: "Scale fluentd deployment config"
+    command: oc scale dc/logging-fluentd --replicas={{ fluentd_replicas | default('1') }}
+    tags:
+      - build
+
+  - name: "Wait for imagestreams to become available"
+    action: shell oc get is | grep logging-fluentd
+    register: result
+    until: result.rc == 0
+    failed_when: result.rc == 1 and 'not found' not in result.stderr
+    retries: 15
+    delay: 5
+    tags:
+      - build
```

Comment:
nm my previous comment, it looks like this should actually be moved above "Set insecured registry"